code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -341,48 +341,26 @@
 
 /obj/machinery/atmospherics/unary/cryo_cell/update_overlays()
 	. = ..()
+	if(occupant_overlay)
+		qdel(occupant_overlay)
 	if(!occupant)
 		. += "lid[on]" //if no occupant, just put the lid overlay on, and ignore the rest
 		return
 
 	if(occupant)
-		var/mutable_appearance/pickle = mutable_appearance(occupant.icon, occupant.icon_state)
-		pickle.overlays = occupant.overlays
-		pickle.pixel_y = LOW
-
-		. += pickle
-		. += "lid[on]"
-		if(on && !running_bob_animation) //no bobbing if off
-			var/bobbing = NONE //used to see if we are going up or down
-			spawn(0) // Without this, the icon update will block. The new thread will die once the occupant leaves.
-				running_bob_animation = TRUE
-				while(occupant && on)
-					cut_overlay("lid[on]") //have to remove the overlays first, to force an update- remove cloning pod overlay
-					cut_overlay(pickle) //remove mob overlay
-
-					switch(pickle.pixel_y) //this looks messy as fuck but it works, switch won't call itself twice
-
-						if(MID) //inbetween state, for smoothness
-							switch(bobbing) //this is set later in the switch, to keep track of where the mob is supposed to go
-								if(UP)
-									pickle.pixel_y = HIGH //set to highest
-
-								if(DOWN)
-									pickle.pixel_y = LOW //set to lowest
-
-						if(LOW) //mob is at it's lowest
-							pickle.pixel_y = MID //set to inbetween
-							bobbing = UP //have to go up
-
-						if(HIGH) //mob is at it's highest
-							pickle.pixel_y = MID //set to inbetween
-							bobbing = DOWN //have to go down
+		occupant_overlay = new(get_turf(src))
+		occupant_overlay.icon = occupant.icon
+		occupant_overlay.icon_state = occupant.icon_state
+		occupant_overlay.overlays = occupant.overlays
+		occupant_overlay.pixel_y = LOW
+		occupant_overlay.layer = layer + 0.01
 
-					add_overlay(pickle) //re-add the mob to the icon
-					add_overlay("lid[on]") //re-add the overlay of the pod, they are inside it, not floating
+		if(on)
+			animate(occupant_overlay, time = 3 SECONDS, loop = -1, easing = BACK_EASING, pixel_y = HIGH)
+			animate(time = 3 SECONDS, loop = -1, easing = BACK_EASING, pixel_y = LOW)
```

Comment:
Doesn't the second animate need an `occupant_overlay` argument too?

Reply:
Animates that follow each other will apply to the same thing if not explicitly targetting something else.

Reply:
Ah good to know.